lib/components/mobile/results-header.js

```diff
@@ -0,0 +1,125 @@
+import LocationIcon from '@opentripplanner/location-icon'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { Button, Col, Row } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+
+import * as formActions from '../../actions/form'
+import * as uiActions from '../../actions/ui'
+import {
+  getActiveItineraries,
+  getActiveSearch,
+  getResponsesWithErrors
+} from '../../util/state'
+
+import MobileNavigationBar from './navigation-bar'
+
+const LocationContainer = styled.div`
+  font-weight: 300;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`
+
+const LocationSummaryContainer = styled.div`
+  height: 50px;
+  left: 0;
+  padding-right: 10px;
+  position: fixed;
+  right: 0;
+  top: 50px;
+`
+
+const LocationsSummaryColFromTo = styled(Col)`
+  font-size: 1.1em;
+  line-height: 1.2em;
+`
+
+const LocationsSummaryRow = styled(Row)`
+  padding: 4px 8px;
+`
+
+const StyledLocationIcon = styled(LocationIcon)`
+  margin: 3px;
+`
+
+/**
+ * This component renders the results header and an error message
+ * if no itinerary was found.
+ */
+class ResultsHeader extends Component {
+  static propTypes = {
+    errors: PropTypes.array,
+    query: PropTypes.object,
+    resultCount: PropTypes.number,
+    setMobileScreen: PropTypes.func
+  }
+
+  _editSearchClicked = () => {
```

Comment:
In batch mode, if you have planned a trip and expanded an itinerary, clicking this takes you back to the search form, but when you re-plan the trip it takes you directly to the expanded itinerary view. Instead, it should take you back to the results list.

Reply:
Good catch, fixed in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/df927a8f852ee3ec599d8f14bdd44406cf9376af.